### PR TITLE
docs: Metrics storybook analysis document

### LINKED ISSUE #1435: Create Storybook analysis
**Description**
Use feature template to create the Analysis section for Storybook feature.

**Acceptance Criteria**
- [ ] create the Analysis section for Storybook feature

**Additional context**
[Feature design template](https://github.com/Flank/flank-dashboard/blob/master/docs/19_feature_design_template.md)

### PR DESCRIPTION
**Pull request by solid-tetianamovlian**
Fixes #1435 

## Checklist

- [x] Documented

### CODE REVIEW

File: metrics/web/docs/features/storybook/01_create_metrics_storybook.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+- The management of themes in these packages is reduced only to the possibility of switching between the dark and the light themes, as well as the presence of a toggle theme button, the behavior of which we cannot control. Even though the packages allow managing themes, it becomes a complex task to integrate the Metrics theme into the storybook. The described above packages do not allow to provide the theme data for all the Metric widgets. See [Metrics Theme guidelines](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#metrics-theme-guidelines) to get more metrics theme implementation details.
+
+- The UI in these packages is, also very basic, without thoughtful design and UX.
+
+Summing up the consideration of approaches and given the above explanations, the best solution would be to create this library from [scratch](#manually-create-a-storybook-from-scratch).
+
```
Comment (solid-vladlaba) [inline metrics/web/docs/features/storybook/01_create_metrics_storybook.md:183]: It seems like the following sections are missing: 
    - [Prototyping](https://github.com/Flank/flank-dashboard/blob/master/docs/19_feature_design_template.md#prototyping)
    - [System modeling](https://github.com/Flank/flank-dashboard/blob/master/docs/19_feature_design_template.md#system-modeling)

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-tetianamovlian @solid-maxim! I've added a couple of suggestions below - please let me know what you think!

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+According to the [widget structure organization document](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#applying-a-theme-to-a-widget-appearance) common widgets use the Metrics theme. Thus to display these widgets in the storybook correctly we should move the Metrics theme along with them. The disadvantage of moving common widgets is the loss of the ability to customize them as base widgets.
+
+Also, as the common Metrics widgets use the Metrics theme, the storybook should have the ability to toggle between the light and dark variants of the theme, to affect widgets' appearance.
+
+Now, we can list the requirements the `Metrics storybook` feature must satisfy:
+
+- Displays a set of Metrics widgets.
+- Provides an ability to manage the Metrics theme.
+- Contains a tab with the widget documentation.
+- Contains a list of inputs to change widgets' appearance (height, width, color, etc.).
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:62]: :readable: 
```suggestion
- Provides an ability to change the widgets' appearance (height, width, color, etc.).
```
What do you think?

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+There are two approaches to create the storybook for Flutter widgets - [using existing packages](#use-existing-packages) or [creating a UI from scratch](#manually-create-a-storybook-from-scratch).
+
+#### Use existing packages
+
+At this time, there are only a few packages, that provide the functionality to write Storybook for Flutter widgets.
+
+Let's take a closer look at the packages:
+
+1. [Storyboard](https://pub.dev/packages/storyboard)
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:79]: :readable: Would it be great to convert this numbered list into separate sub-sections?

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+Another solution is to provide a completely new Flutter project with the [required functionality](#requirements).
+
+Pros:
+
+- Allows creating a custom Storybook with all required features.
+
+Cons:
+
+- Requires designing a new UI for the Metrics storybook.
+- Takes extra time to implement a new UI element.
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:163]: 🏷️ 
```suggestion
- Takes extra time to implement new UI elements.
```
?

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+- The management of themes in these packages is reduced only to the possibility of switching between the dark and the light themes, as well as the presence of a toggle theme button, the behavior of which we cannot control. Even though the packages allow managing themes, it becomes a complex task to integrate the Metrics theme into the storybook. The described above packages do not allow to provide the theme data for all the Metric widgets. See [Metrics Theme guidelines](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#metrics-theme-guidelines) to get more metrics theme implementation details.
+
+- The UI in these packages is, also very basic, without thoughtful design and UX.
+
+Summing up the consideration of approaches and given the above explanations, the best solution would be to create this library from [scratch](#manually-create-a-storybook-from-scratch).
+
+### Prototyping
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:177]: 🔨 I think the prototyping section should contain information that confirms that the feature implementation is possible. For example, we can prove that we can extract the part of the Metrics Web application into a separate package by providing appropriate code snippets. What do you think?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-maxim & @solid-tetianamovlian! I've added a couple of suggestions here - please take a look.

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+First of all, the great purpose of the Metrics storybook is that each widget and story (a specific state of a widget) can be easily reused across the entire project by any developer who works on it, ensuring consistent design and UX. Metrics storybook creating allows developers to work in parallel on widgets and do not touch the main Metrics Web Application's codebase. It also, makes it easier to add new widgets, since we can test them separately and only then make changes to the code of the application.
+
+The storybook application can be shared with and commented on by designers so that they can add their input regarding the implementation of the designs.
+
+Another group that can use this feature is clients. Storybook makes it easy to show pieces of the software to keep the client in the loop. Even small pieces of UI can be shared and made available for feedback, preventing longer periods without any deliverables.
+
+### Requirements
+> Define requirements and make sure that they are complete.
+
+Before defining requirements to the feature, we should determine which components from the Metrics Web Application should be included in the storybook.
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:47]: :tw:
```suggestion
Before defining requirements to the feature, we should determine which components from the Metrics Web Application the storybook should include.
```
Please, prefer active voice to passive voice. See [Google Technical Writing](https://developers.google.com/tech-writing/one/active-voice#prefer_active_voice_to_passive_voice) for more details.

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+
+At the same time, some of the common Metrics widgets (e.g., [MetricsUserMenu](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/lib/common/presentation/widgets/metrics_user_menu.dart)) are using [ChangeNotifiers](https://pub.dev/documentation/provider/latest/provider/ChangeNotifierProvider-class.html) internally. It makes widgets dependent on the Metrics Web Application's notifiers, so in these cases, we cannot extract this kind of widget, without changing them.
+
+According to the [widget structure organization document](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#applying-a-theme-to-a-widget-appearance) common widgets use the Metrics theme. Thus to display these widgets in the storybook correctly, we should move the Metrics theme along with them. The disadvantage of moving common widgets is the loss of the ability to customize them as base widgets.
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:57]: 🏷️
```suggestion
According to the [widget structure organization document](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/03_widget_structure_organization.md#applying-a-theme-to-a-widget-appearance), common widgets use the Metrics theme. Thus to display these widgets in the storybook correctly, we should move the Metrics theme along with them. The disadvantage of moving common widgets is the loss of the ability to customize them as base widgets.
```

File: metrics/web/docs/features/storybook/01_metrics_storybook_design.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+
+#### Use existing packages
+
+At this time, there are only a few packages, that provide the functionality to write Storybook for Flutter widgets.
+
+Let's take a closer look at the packages:
+
+#### [Storyboard](https://pub.dev/packages/storyboard)
+
+There are simple classes, which provide all the logic to start writing the storybook. It is just a little package, that helps to reduce a boilerplate.
```
Comment (solid-vovabeloded) [inline metrics/web/docs/features/storybook/01_metrics_storybook_design.md:85]: 🏷️ 
```suggestion
There are simple classes, which provide all the logic to start writing the storybook. It is just a little package that helps to reduce a boilerplate.
```

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Well done, @solid-maxim & @solid-tetianamovlian! I've added a few more minor suggestions here - please take a look.

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-maxim & @solid-tetianamovlian!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-maxim & @solid-tetianamovlian!

**Pull request merged**